test/helpers/position_assertions.cc

```diff
@@ -1130,102 +1130,271 @@ void ApplyCodeActionAssertion::check(const UnorderedMap<std::string, std::shared
+                    << fmt::format("Improperly formatted `symbol-search` assertion.\n"
+                                   "Expected `rank = integer`, got `{}` in:\n{}\n.",
+                                   optionPair, assertionContents);
+            }
+            rank = atoi(matches[5].str().c_str());
+        } else if (optionName == "uri") {
+            if (!matches[4].matched) {
+                ADD_FAILURE_AT(string(filename).c_str(), assertionLine + 1)
+                    << fmt::format("Improperly formatted `symbol-search` assertion.\n"
+                                   "Expected `uri = \"substr\"`, got `{}` in:\n{}\n.",
+                                   optionPair, assertionContents);
+            }
+            uri = matches[4].str();
+        } else {
+            ADD_FAILURE_AT(string(filename).c_str(), assertionLine + 1)
+                << fmt::format("Improperly formatted `symbol-search` assertion for query {}.\n"
+                               "Valid args are name, container, rank, and uri, could not parse `{}` in:\n{}\n",
+                               query, optionPair, assertionContents);
+        }
+        remainingOptions = matches[6].str();
     }
+    return make_shared<SymbolSearchAssertion>(filename, range, assertionLine, query, name, container, rank, uri);
+}
 
-    ADD_FAILURE_AT(string(filename).c_str(), assertionLine + 1) << fmt::format(
-        "Improperly formatted assertion. Expected 'symbol-search: \"<query>\"'. Found '{}'", assertionContents);
+bool SymbolSearchAssertion::matches(std::string_view uriPrefix, const SymbolInformation &symbol) const {
+    auto assertionLocation = getLocation(uriPrefix);
+    auto &symbolLocation = symbol.location;
+    if (uri.has_value()) {
+        if (symbolLocation->uri.find(*uri) == string::npos) {
+            return false;
+        }
+    } else {
+        if (assertionLocation->uri != symbolLocation->uri) {
+            return false;
+        }
+        if (assertionLocation->range->start->line != symbolLocation->range->start->line) {
+            return false;
+        }
+    }
+    if (name.has_value() && *name != symbol.name) {
+        return false;
+    }
+    if (container.has_value() && *container != symbol.containerName) {
+        return false;
+    }
+    return true;
+}
+namespace {
+string pluralized_count(string_view word, int count) {
+    return fmt::format("{} {}{}", count, word, (count == 1) ? "" : "s");
+}
 
-    return nullptr;
+void addFailureAtLocationWithSource(const Location &location,
+                                    const UnorderedMap<string, shared_ptr<core::File>> &sourceFileContents,
+                                    string_view uriPrefix, std::string_view message) {
+    if (!isUriATestFile(uriPrefix, location.uri)) {
+        ADD_FAILURE_AT(location.uri.c_str(), location.range->start->line + 1)
+            << fmt::format("{}\n(source unavailable for {})\n", message, location.uri);
```

Comment:
In which cases is the source unavailable?

Reply:
For results from the `stdlib` (i.e., `rbi` files).